pimd/pim_neighbor.c

```diff
@@ -362,19 +362,16 @@ static void delete_prefix_list(struct pim_neighbor *neigh)
 #ifdef DUMP_PREFIX_LIST
 		struct listnode *p_node;
 		struct prefix *p;
-		char addr_str[10];
 		int list_size = neigh->prefix_list
 					? (int)listcount(neigh->prefix_list)
 					: -1;
 		int i = 0;
 		for (ALL_LIST_ELEMENTS_RO(neigh->prefix_list, p_node, p)) {
-			pim_inet4_dump("<addr?>", p->u.prefix4, addr_str,
-				       sizeof(addr_str));
 			zlog_debug(
-				"%s: DUMP_PREFIX_LIST neigh=%x prefix_list=%x prefix=%x addr=%s [%d/%d]",
+				"%s: DUMP_PREFIX_LIST neigh=%x prefix_list=%x prefix=%x addr=%pFX [%d/%d]",
```

Comment:
btw, previously it was printed as an address, not as a prefix, is this intentional?

Reply:
Its a `pimd` idiosyncracy of weird design…

But it should use `%pFXh` since the struct prefix is used as an address.

Reply:
Done